src/Gui/Selection/Selection.cpp

```diff
@@ -311,6 +311,101 @@ std::vector<SelectionSingleton::SelObj> SelectionSingleton::getPickedList(const
     return temp;
 }
 
+std::vector<Gui::SelectionObject> SelectionSingleton::getSelectionIn(App::DocumentObject* container,
+    Base::Type typeId, bool single) const
+{
+    if (!container) {
+        return getSelectionEx(nullptr, typeId, ResolveMode::NoResolve, single);
+    }
+
+    std::vector<SelectionObject> sels = getSelectionEx(nullptr, App::DocumentObject::getClassTypeId(), ResolveMode::NoResolve, single);
+
+    std::vector<SelectionObject> ret;
+    std::map<App::DocumentObject*, size_t> SortMap;
+
+    for (auto& sel : sels) {
+        auto* rootObj = sel.getObject();
+        App::Document* doc = rootObj->getDocument();
+        std::vector<std::string> subs = sel.getSubNames();
+        bool objPassed = false;
+
+        for (size_t i = 0; i < subs.size(); ++i) {
+            auto& sub = subs[i];

```

Comment:
```suggestion
        for (const std::string& sub : subs) {
```
Why not use the nice syntax above here as well?

Reply:
i is needed. And when it is the case I prefer this syntax

Reply:
I agree